src/diffpy/srmise/tests/test_dataclusters.py

```diff
@@ -32,3 +33,86 @@ def test___eq__():
             print(f"not-equal test failed on {attr_key}")
             assert not expected == actual
         attributes.update({attr_key: reset})
+
+    # In the set data test, we test for these cases.
+    # (1) x and y are non-empty array values, and res is positive (the most generic case)
+    # (2) x and y are non-empty array values, and res is 0 (will produce a msg that makes trivial clustering)
+    # (3) x and y are non-empty array values, and res is negative (will produce a ValueError,
+    # msg = please enter a non-negative res value)
+    # (4, 5) One of x and y is empty array, and res is positive
+    # (produce ValueError & msg "Sequences x and y must have the same length.", something like that)
+    # (6) Both x and y are empty array, and res is zero.
+
+
+@pytest.mark.parametrize(
+    "inputs, expected",
+    [
+        (
+            # case (1)
+            {
+                "input_x": np.array([1, 2, 3]),
+                "input_y": np.array([3, 2, 1]),
+                "input_res": 4,
+            },
+            DataClusters(np.array([1, 2, 3]), np.array([3, 2, 1]), 4),
+        ),
+        (
+            # case (6)
+            {
+                "input_x": np.array([]),
+                "input_y": np.array([]),
+                "input_res": 0,
+            },
+            DataClusters(np.array([]), np.array([]), 0),
+        ),
+    ],
+)
+def test_set_data(inputs, expected):
+    actual = DataClusters(x=inputs["input_x"], y=inputs["input_y"], res=inputs["input_res"])
+    assert actual == expected
```

Comment:
just a note, the expected we would like to explicitly set all its attributes.  this will also implicitly test other parts of the code that generate those attributes.